application/src/main/java/org/thingsboard/server/service/mail/TbMailSender.java

```diff
@@ -71,24 +70,27 @@ public TbMailSender(TbMailContextComponent ctx, JsonNode jsonConfig) {
         setJavaMailProperties(createJavaMailProperties(jsonConfig));
     }
 
-    @SneakyThrows
     @Override
     public void doSend(MimeMessage[] mimeMessages, @Nullable Object[] originalMessages) {
-        if (oauth2Enabled && (System.currentTimeMillis() > tokenExpires)){
-            refreshAccessToken();
-            setPassword(accessToken);
-        }
+        updateOauth2PasswordIfExpired();
         super.doSend(mimeMessages, originalMessages);
     }
 
-    @SneakyThrows
     @Override
-    public void testConnection() {
+    public void testConnection() throws MessagingException {
```

Comment:
Please, do a simple test to ensure that  updateOauth2PasswordIfExpired() is called once when testConnection()
use mocks wherever is possible